addons/l10n_in/tests/test_l10n_in_fiscal_position.py

```diff
@@ -132,3 +132,87 @@ def test_l10n_in_fiscal_in_bill_to_ship_to(self):
             out_invoice.fiscal_position_id,
             self.env['account.chart.template'].ref('fiscal_position_in_export_sez_in')
         )
+
+    def test_l10n_in_fiscal_in_vendor_bills(self):
+        '''
+        In Purchase Document: Compare place of supply with vendor state
+        '''
+
+        self.env.company = self.default_company
+        template = self.env['account.chart.template']
+        company_state = self.env.company.state_id
+        other_state = self.env['res.country.state'].search([
+            ('id', '!=', company_state.id),
+            ('country_id', '=', company_state.country_id.id)],
+            limit=1
+        )
+
+        # Sub-test: Intra-State
+        with self.subTest(scenario="Intra-State"):
+            self.partner_a.write({'state_id': company_state.id})
+            vendor_bill = self._assert_invoice_fiscal_position(
+                fiscal_position_ref='fiscal_position_in_intra_state',
+                partner=self.partner_a,
+                move_type='in_invoice',
+                post=False,
+            )
+            self.partner_a.write({'state_id': other_state.id})
+            vendor_bill.write({'l10n_in_state_id': other_state.id})
+            self.assertEqual(
+                vendor_bill.fiscal_position_id,
+                template.ref('fiscal_position_in_intra_state')
+            )
+
+        # Sub-test: Inter-State
+        with self.subTest(scenario="Inter-State"):
+            self.partner_a.write({'state_id': other_state.id})
+            vendor_bill = self._assert_invoice_fiscal_position(
+                fiscal_position_ref='fiscal_position_in_inter_state',
+                partner=self.partner_a,
+                move_type='in_invoice',
+                post=False,
+            )
+            self.partner_a.write({'state_id': company_state.id})
+            vendor_bill.write({'l10n_in_state_id': other_state.id})
+            self.assertEqual(
+                vendor_bill.fiscal_position_id,
+                template.ref('fiscal_position_in_inter_state')
+            )
+
+        # Sub-test: Export/SEZ (Outside India)
+        with self.subTest(scenario="Export/SEZ"):
+            vendor_bill = self._assert_invoice_fiscal_position(
+                fiscal_position_ref='fiscal_position_in_export_sez_in',
+                partner=self.partner_foreign,
+                move_type='in_invoice',
+                post=False,
+            )
+            self.partner_a.write({'state_id': company_state.id})
+            vendor_bill.write({'l10n_in_state_id': self.env.ref('l10n_in.state_in_oc').id})  # Other Country State
+            self.assertEqual(
+                vendor_bill.fiscal_position_id,
+                template.ref('fiscal_position_in_export_sez_in')
+            )
+            # Here fpos should Inter-State. But due to `l10n_in_gst_treatment` it will be Export/SEZ
```

Comment:
```suggestion
            vendor_bill.write({'l10n_in_state_id': self.env.ref('l10n_in.state_in_oc').id})  # Other Country State
            self.assertEqual(
                vendor_bill.fiscal_position_id,
                template.ref('fiscal_position_in_export_sez_in')
            )
            # Here fpos should Inter-State. But due to `l10n_in_gst_treatment` it will be Export/SEZ
            self.partner_a.write({'state_id': company_state.id})
```